core-text/src/font.rs

```diff
@@ -1006,9 +1009,41 @@ fn out_of_range_variations() {
                 .unwrap()
                 .to_f64()
                 .unwrap();
-            assert_eq!(val, max + 1.);
+
+            assert_eq!(val, max, "axis: {:?} = {:?} (not {:?})", tag, val, max);
+        }
+    } else if version >= (12, 0, 0) && version < (13, 0, 0) {
+        let var_attrs = var_attrs.unwrap().downcast::<CFDictionary>().unwrap();
+        assert!(!var_attrs.is_empty());
+        let var_attrs: CFDictionary<CFType, CFType> = unsafe { std::mem::transmute(var_attrs) };
+        // attributes greater than max remain
+        for axis in axes.iter() {
+            let tag = axis
+                .find(unsafe { kCTFontVariationAxisIdentifierKey })
+                .unwrap();
+            let max = axis
+                .find(unsafe { kCTFontVariationAxisMaximumValueKey })
+                .unwrap()
+                .downcast::<CFNumber>()
+                .unwrap()
+                .to_f64()
+                .unwrap();
+            let val = var_attrs
+                .find(tag.clone())
+                .unwrap()
+                .downcast::<CFNumber>()
+                .unwrap()
+                .to_f64()
+                .unwrap();
+
+            let expected = max + 1.;
+            assert_eq!(
+                val, expected,
+                "axis: {:?} = {:?} (not {:?})",
+                tag, val, expected
```

Comment:
It's pretty hard to tell from this change, but is the only difference the lack of the assertion? If that's the case, I just add the conditional around the assertion and add a comment explaining why it doesn't apply.

Reply:
I split my changes so they're hopefully easier to review on a commit-by-commit basis, and extracted a function with a parameter instead of duplicating this.

Hope that helps.